ixdiagnose/plugins/system_state.py

```diff
@@ -0,0 +1,58 @@
+import contextlib
+import json
+
+from ixdiagnose.utils.command import Command
+from ixdiagnose.utils.run import run
+
+from .base import Plugin
+from .metrics import CommandMetric, FileMetric, PythonMetric
+
+
+def get_ds_list() -> list:
+    ds_list = []
+    file_path = '/conf/truenas_root_ds.json'
+
+    with contextlib.suppress(FileNotFoundError, json.JSONDecodeError):
+        with open(file_path, 'r') as file:
+            data = json.load(file)
+
+        for entry in data:
+            if entry.get('fhs_entry', {}).get('snap') and entry.get('ds'):
+                ds_list.append(entry['ds'])
+
+    return ds_list
+
+
+def get_root_ds(client: None, resource_type: str) -> str:
```

Comment:
Let's make this serializable because the solutions team is making bugclerk automatically parse the debug archives attached to tickets and it will print a summary. This is something that can be added to their checks.